packages/flutter/lib/src/widgets/magnifier.dart

```diff
@@ -0,0 +1,545 @@
+  ///           builder: (BuildContext context) => const Text('I WILL display in the magnifier')));
+  /// }
+  /// ```
+  /// {@end-tool}
+  ///
+  /// A null check on [overlayEntry] will not suffice to check if a magnifier is in the
+  /// overlay or not; instead, you should check [shown]. This is because it is possible,
+  /// such as in cases where [hide] was called with `removeFromOverlay` false, that the magnifier
+  /// is not shown, but the entry is not null.
+  OverlayEntry? overlayEntry;
+
+  /// If the magnifier is shown or not.
+  ///
+  /// [shown] is:
+  /// - false when nothing is in the overlay.
+  /// - false when [animationController] is [AnimationStatus.dismissed].
+  /// - false when [animationController] is animating out.
+  /// and true in all other circumstances.
+  bool get shown {
+    if (overlayEntry == null) {
+      return false;
+    }
+
+    if (animationController != null) {
+      return animationController!.status == AnimationStatus.completed ||
+          animationController!.status == AnimationStatus.forward;
+    }
+
+    return true;
+  }
+
+  /// Shows the [RawMagnifier] that this controller controls.
+  ///
+  /// Returns a future that completes when the magnifier is fully shown, i.e. done
+  /// with its entry animation.
+  ///
+  /// To control what overlays are shown in the magnifier, utilize [below]. See
+  /// [overlayEntry] for more details on how to utilize [below].
+  ///
+  /// If the magnifier already exists (i.e. [overlayEntry] != null), then [show] will
+  /// reshow the old overlay.
+  Future<void> show({
+    required BuildContext context,
+    required WidgetBuilder builder,
+    Widget? debugRequiredFor,
+    OverlayEntry? below,
+  }) async {
+    if (overlayEntry != null) {
+      if (animationController?.status == AnimationStatus.dismissed) {
+        await animationController!.forward();
+      }
+
+      return;
+    }
+
+    final OverlayState? overlayState = Overlay.of(
+      context,
+      rootOverlay: true,
+      debugRequiredFor: debugRequiredFor,
+    );
+
+    final CapturedThemes capturedThemes = InheritedTheme.capture(
+      from: context,
+      to: Navigator.maybeOf(context)?.context,
+    );
+
+    overlayEntry = OverlayEntry(
+      builder: (BuildContext context) => capturedThemes.wrap(builder(context)),
+    );
+    overlayState!.insert(overlayEntry!, below: below);
+
+    if (animationController != null) {
+      await animationController?.forward();
+    }
+  }
+
+  /// Schedules a hide of the magnifier.
+  ///
+  /// If this [MagnifierController] has an [AnimationController],
+  /// then [hide] reverses the animation controller and waits
+  /// for the animation to complete. Then, if [removeFromOverlay]
+  /// is true, remove the magnifier from the overlay.
+  ///
+  /// In general, [removeFromOverlay] should be true, unless
+  /// the magnifier needs to preserve states between shows / hides.
+  Future<void> hide({bool removeFromOverlay = true}) async {
```

Comment:
Nice feature. Do we think there might be a need for a version of this that hides the magnifier immediately (i.e. abort any animation and remove the overlay entry directly)?

Reply:
I made one that was visible for testing (per longcatisloong) and I'll leave it private. If we ever need to expose it, it'll be a quick change